wiki/docs/developer-resources/functional-spec.md

```diff
@@ -0,0 +1,218 @@
+2. **Name**: the unique name of the satellite
+3. **Microphone device**: the entity to use as the microphone device. This devices captures the intent of the user speaking to the satellite.
+4. **Media player device**: the entity to use as the mediaplayer for the satellite. TTS output will be sent to this media player device.
+5. **Music player device**: the entity to use as the musicplayer for the satellite. This can be different or the same as the media player device. Will be used to play music to.
+6. **Display device**: which entity to use to display satellite output on. Only available for satellites that are configured as type display. This should be set to the browser path entity of browser mod for the satellite.
+7. **Browser ID**: the ID of the browser mod installation. Only available for satellites that are configured as type display. 
+
+### Options flow
+The options flow is available after finishing the config flow for a satellite and will give the user access to the following settings so they can change these without having to delete and re-create their satellite config:
+1. Type
+2. Microphone device
+3. Media player device
+4. Music player device
+5. Display device
+6. Browser ID
+7. Specific weather entity to use (to override weather entity configured for all satellites)
+
+These settings are in the options flow because we don't feel like the user needs to change them that often after they are set up (principle ##5).
+
+## What the integration set up creates
+Succesfully completing at least one config flow of the integration creates the following items:
+- **a sensor** representing each satellite. Each satellite has its own sensor entity
+- **a panel** that hosts the views. There is only one panel and all satellites share the same panel but can have specific configurations for the views in the panel (see below).
+- **a group** hosting all the satellites. There is only one group of satellites.
+- **a timer** helper for each satellite. there is only one timer per satellite.
+- **services** that can be used to cause specific actions to happen on the satellites
+
+## Services
+The integration provides the following services:
+
+1. **Broadcast service**. This sends a message to all satellites which is shown on the intent view.
+1. **Set view service** (for all satellites or specific satellites). This switches the view on the specified satellites.
+1. **Set mode service**  (for all satellites or specific satellites). This sets the mode on the specified satellites.
+1. **Set do not disturb** service (for all satellites or specific satellites). This sets the specified sattelites to do not disturb or turns off do not disturb.
+
+## Panel
+This is the big one. The panel serves multiple purposes:
+-  It's the dashboard satellites with a display will show. This will be configured automatically as part of the config flow for the satellite.
+-  The dashboard consists of multiple views. The following views are provided:
```

Comment:
As a starting point, I think we should go for:

- Clock
- Alarm
- Intent
- Weather

Keep the deliverables small so we can get some users testing it out and providing feedback, then we know what the next requirements will be. Pretty much like what HA does with new features.